### PR TITLE
Improve Item Collision

### PR DESCRIPTION
**Pull request by MarkSedhom1005166721**
## Changes:
- Now when we pick up items, it's physics based, so it'll follow you around and bump into stuff on the way (main character has the force now!)
- All cups (except ruffian and old lady cups) can be picked up and thrown around
- Moved question mark to on top 

## Comments:
- Sometimes movement of items is still jittery. I have a few ideas to fix it, but I think I've spent too long on this and that could be done in polish stage.

## Screenshots:

https://user-images.githubusercontent.com/77755875/198835768-8774b000-1c4b-4be4-bbe5-d9e6061e07ac.mp4

### How to Test:
Pick up items and run around with them

### Sample Test Cases:
Try your absolute best to get the items stuck

### Impacted Features:
Interactables that get picked up, and cups

### CODE REVIEW

**Comment by MarkSedhom1005166721** [comment]
I mentioned this on the discord, but will mention here again. Only files changed are Item.cs and ItemHolder.cs. All other 'changed files' are due to me merging in PR 36

File: Assets/Scripts/Character/ItemHolder.cs [hunk 1]
```diff
@@ -7,17 +7,67 @@ public class ItemHolder : MonoBehaviour {
...
+
   private void IgnoreCollisions(Item item, bool ignore) {
-    foreach (Collider collider in item.GetComponentsInChildren<Collider>()) {
+    foreach (Collider collider in item.allColliders) {
       Physics.IgnoreCollision(holderCollider, collider, ignore);
     }
   }
 
+  private void ForceToHand() {
+    if (_heldItem && _heldItem.Rigidbody) {
```
Comment (c-boyle) [inline Assets/Scripts/Character/ItemHolder.cs:39]: Nitpick: I like `_heldItem != null` better than just `_heldItem` :P feels more clear imo

File: Assets/Scripts/Item.cs [hunk 2]
```diff
@@ -29,9 +38,19 @@ public class Item : MonoBehaviour {
...
       children.Add(childTransform.gameObject);
     }
+    allColliders = new List<Collider>(GetComponentsInChildren<Collider>());
+
+    itemBounds = allColliders[0].bounds;
+    for(int i = 1; i < allColliders.Count; i++){
+      itemBounds.Encapsulate(allColliders[i].bounds);
+    }
+
+    structureBounds = GameObject.Find("Structure").GetComponent<Collider>().bounds;
```
Comment (c-boyle) [inline Assets/Scripts/Item.cs:53]: I think adding a collider field that's also exposed in the inspector for this "Structure" collider would be a bit more efficient / harder to accidentally forget when setting up an item object.

**Review by c-boyle — APPROVED** [review]

File: Assets/Scripts/Item.cs (hunk 2 above)
Comment (MarkSedhom1005166721) [reply Assets/Scripts/Item.cs:53]: Oh honestly I forgot that was there. I no longer use the structureBounds variable. I'll just remove it

**Pull request merged**